src/dune_engine/action.ml

```diff
@@ -308,3 +308,11 @@ let is_useful_to_sandbox = is_useful_to false false
 let is_useful_to_distribute = is_useful_to true false
 
 let is_useful_to_memoize = is_useful_to true true
+
+module Full = struct
```

Comment:
I saw this change while reviewing a subsequent PR (while it was unrebased) and thought that the name `Full` is a bit too generic. I would call it `For_execution` or something similar to make it a bit more self-describing.

Reply:
I would argue that any action is there for execution, and this name is longer, so I like that name less. I could see "With_env" being a reasonable name, but honestly I don't see what confusion can arise from the current name.

Reply:
Every action is destined for execution but the lives of some are spared, e.g. if their results are available in the cache :)

Now that I wrote that, I realised that even the actions whose results are restored from the cache must be `Full`, so my initial name doesn't quite work.

Anyway, we can certainly live with `Full`.